### PR TITLE
[#8] [Backend] As a user, I can see the survey questions

### LINKED ISSUE #8: [Backend] As a user, I can see the survey questions
## Why

When a user starts taking a survey, he/she will go through a set of questions of the selected survey. The mobile app should show how many questions there are for the survey so that the user can expect how long it is going to take and he/she can always stop at any point by dismissing the screen.

## Acceptance Criteria

- Call `GET /api/v1/surveys/{id}` endpoint with the following query parameter:
  - `id`: Survey's id

## Resources

[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#142b8597-e176-4e74-95a5-cb48ed5115f0)

### PR DESCRIPTION
**Pull request by Wadeewee**
#8

## What happened 👀

- Call `GET /api/v1/surveys/{id}` endpoint with the following query parameter:
  - `id`: Survey's id

## Insight 📝

- Create response classes: `SurveyAnswerResponse`, `SurveyQuestionResponse`, and `SurveyDetailResponse`.
- Create model classes: `SurveyAnswerModel`, `SurveyQuestionModel`, and `SurveyDetailModel`.
- Create `GetSurveyDetailUseCase` and `GetSurveyDetailUseCaseTest`.
- Add more method `getSuyveyDetail` in `SurveyRepository`.
- Remove unnecessary classes.
 
## Proof Of Work 📹

```
flutter: *** Request ***
flutter: uri: https://survey-api.nimblehq.co/api/v1/surveys/d5de6a8f8f5f1cfe51bc
flutter: method: GET
flutter: responseType: ResponseType.json
flutter: followRedirects: true
flutter: persistentConnection: true
flutter: connectTimeout: 0:00:03.000000
flutter: sendTimeout: null
flutter: receiveTimeout: 0:00:05.000000
flutter: receiveDataWhenStatusError: true
flutter: extra: {}
flutter: headers:
flutter:  Content-Type: application/json; charset=utf-8
flutter:  Authorization: Bearer nPIrU8eYzBusoP1SsQAvLv_MsBOQjbOBNEpzhiRP8eQ
flutter: data:
flutter: null
flutter:
flutter: *** Response ***
flutter: uri: https://survey-api.nimblehq.co/api/v1/surveys/d5de6a8f8f5f1cfe51bc
flutter: statusCode: 200
flutter: headers:
flutter:  connection: keep-alive
flutter:  cache-control: max-age=0, private, must-revalidate
flutter:  date: Fri, 26 May 2023 07:11:40 GMT
flutter:  vary: Accept-Encoding, Origin
flutter:  content-encoding: gzip
flutter:  referrer-policy: strict-origin-when-cross-origin
flutter:  report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=IuZIoNNbVyJR3FU112BFFwiz9nAjivRQ1w8KAOowuCirqejqR8J%2F7tjxpy6mqtfl3FbLyWV2k5OFlFUnJgUaLZJp8c8%2BpuLyMSufQX%2B3ed4Rz%2B7CzybW0Cl%2FrWGUAzkg4Yp%2BvuotPoxS"}],"group":"cf-nel","max_age":604800}
flutter:  cf-cache-status: DYNAMIC
flutter:  x-permitted-cross-domain-policies: none
flutter:  content-type: application/json; charset=utf-8
flutter:  x-xss-protection: 0
flutter:  server: cloudflare
flutter:  x-request-id: 9fb5ec04-3169-40e4-8eeb-02358d2b9ef0
flutter:  alt-svc: h3=":443"; ma=86400
flutter:  content-length: 3985
flutter:  nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
flutter:  x-download-options: noopen
flutter:  cf-ray: 7cd434918afe10fc-BKK
flutter:  x-runtime: 0.022261
flutter:  etag: W/"d1e76ff303b815bae02a99b4e3f3a69a"
flutter:  via: 1.1 vegur
flutter:  x-frame-options: SAMEORIGIN
flutter:  x-content-type-options: nosniff
flutter: Response Text:
flutter: {"data":{"id":"d5de6a8f8f5f1cfe51bc","type":"survey","attributes":{"title":"Scarlett Bangkok","description":"We'd love ot hear from you!","thank_email_above_threshold":"<span style=\"font-family:arial,helvetica,sans-serif\"><span style=\"font-size:14px\">Dear {name},<br /><br />Thank you for visiting Scarlett Wine Bar &amp; Restaurant at Pullman Bangkok Hotel G &nbsp;and for taking the time to complete our guest feedback survey!<br /><br />Your feedback is very important to us and each survey is read individually by the management and owners shortly after it is sent. We discuss comments and suggestions at our daily meetings and use them to constantly improve our services.<br /><br />We would very much appreciate it if you could take a few more moments and review us on TripAdvisor regarding your recent visit. By <a href=\"https://www.tripadvisor.com/Restaurant_Review-g293916-d2629404-Reviews-Scarlett_Wine_Bar_Restaurant-Bangkok.html\">clicking here</a> you will be directed to our page.&<…>
flutter:

```

### CODE REVIEW

File: lib/model/survey_detail_model.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  final String id;
+  final List<SurveyQuestionModel> questions;
+
+  const SurveyDetailModel({
+    required this.id,
+    required this.questions,
+  });
+
+  @override
+  List<Object?> get props => [questions];
```
Comment (manh-t) [inline lib/model/survey_detail_model.dart]: @Wadeewee we should put all variables to `props` 🙋 
```suggestion
  List<Object?> get props => [id, questions];
```

**Review by manh-t — COMMENTED** [review]
rest LGTM

File: lib/api/response/survey_question_response.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@JsonSerializable()
+class SurveyQuestionResponse {
+  final String id;
+  final String? text;
+  final int? displayOrder;
+  final String? displayType;
+  final String? imageUrl;
+  final String? coverImageUrl;
+  final double? coverImageOpacity;
+  final List<SurveyAnswerResponse> answers;
```
Comment (hoangnguyen92dn) [inline lib/api/response/survey_question_response.dart]: Should we define the list of answers nullable as well?

**Review by hoangnguyen92dn — APPROVED** [review]
The rest LGTM 👍

File: lib/model/survey_detail_model.dart (hunk 1 above)
Comment (Wadeewee) [reply lib/model/survey_detail_model.dart]: Good catch @manh-t 👍

File: lib/api/response/survey_question_response.dart (hunk 2 above)
Comment (Wadeewee) [reply lib/api/response/survey_question_response.dart]: Yup, @hoangnguyen92dn To be safe I think It's good to define nullable here 👍

**Review by hoangnguyen92dn — APPROVED** [review]

**Review by manh-t — APPROVED** [review]

**Pull request merged**